### PR TITLE
usb-reset: init at 0.3

### PR DESCRIPTION
**Pull request by evils**
###### Motivation for this change
i wanted to try out the utility

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
  - not a lot of devices seem to respond, but i got some serial interface reappearing after issuing a reset with this
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Kranzes** [comment]
1. I am almost sure you can get the same functionally of this program without it.
2. If you just wanted to try this utility, can't you just use it first on your local system and build it with nix-build?

File: pkgs/applications/misc/usb-reset/default.nix [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+  buildInputs = [ libusb1 ];
+
+  postPatch = ''
+    substituteInPlace Makefile \
+      --replace /usr/include/libusb-1.0 \
+        ${libusb1.dev}/include/libusb-1.0
+  '';
+
+  makeFlags = [ "DESTDIR=${placeholder "out"}" "prefix=" ];
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/usb-reset/default.nix:29]:
```suggestion
      --replace /usr/include/libusb-1.0 ${libusb1.dev}/include/libusb-1.0
  '';

  makeFlags = [
    "DESTDIR=${placeholder "out"}"
    "prefix="
  ];
```

**Comment by evils** [comment]
@Kranzes 
if you can tell me how to get that functionality without this, please do
(just about everything i've found so far involves either compiling code without the value of the author having packaged that
or requires the use of the device path rather than the IDs as this does)

i did build and run it locally,
this PR is a side effect of that :)

File: pkgs/applications/misc/usb-reset/default.nix (hunk 1 above)
Comment (evils) [reply pkgs/applications/misc/usb-reset/default.nix]: just a note:
while you can get away with this on something as trivial as this
forcing changes onto my branch and leaving me <1h to object to that
feels a bit authoritarian

(i'd have taken and squashed your suggestion if i was available at the time)

**Pull request merged**